tests/ansible_vagrant/Vagrantfile

```diff
@@ -0,0 +1,65 @@
+IMAGE_NAME = "ubuntu/bionic64"
+BRIDGE_NIC = "enp1s0f0"
+
+Vagrant.configure("2") do |config|
+    config.ssh.insert_key = true
+
+    config.vm.provider "virtualbox" do |v|
+        v.memory = 4096
+        v.cpus = 2
+    end
+      
+    config.vm.define "k8s-master" do |master|
+        master.vm.box = IMAGE_NAME
+        # change the bridge interface to match the one on your host machine
+        master.vm.network "public_network", bridge: BRIDGE_NIC, ip: "192.168.0.10"
+        master.vm.hostname = "k8s-master"
+        master.vm.provision "ansible" do |ansible|
+            # Configures the ssh-key
+            ansible.playbook = "kubernetes-playbooks/vagrant-ssh-key.yml"
+             ansible.extra_vars = {
+                pub_key_path: "/home/aldo/.ssh/vagrant_machines.pub"
```

Comment:
Why are there references to your local file system?

Reply:
the Vagrant file is generated by the playbook. you just need to change the variables in group_vars/all/vars.yml.

Reply:
If it is generated dynamically, why is it in the git repository?

Reply:
i forgot to put it in gitignore. with the new commit i resolved this issue, and implemented the features and fixes suggested by @michelsciortino.